src/bloqade/squin/groups.py

```diff
@@ -18,7 +20,10 @@ def run_pass(method: ir.Method, *, fold=True, typeinfer=True):
 
         if typeinfer:
             typeinfer_pass(method)
+
         ilist_desugar_pass(method)
+        py_mult_to_mult_pass(method)
```

Comment:
oh wait this is before type inference. I guess something is wrong in the type annotation of Scale then.